src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -123,19 +123,16 @@ private void ifUsernameAlreadyExistsThrowException(String username) {
     }
   }
 
-  private User findUserToActivateOrThrowException(UUID userId, UUID activationTokenUUID) {
-    Token activationToken = tokenRepository.findByTokenId(activationTokenUUID)
-            .orElseThrow(() -> new IllegalArgumentException("Token is invalid or user is already activated"));
-      return activationToken.getUser();
+  private User findUserToActivateOrThrowException(UUID activationTokenUUID) {
+    UUID userUUID = tokenService.findUser(activationTokenUUID);
+    return userRepository.findByUserIdAndActivatedIsFalse(userUUID)
+            .orElseThrow(() -> new IllegalArgumentException("Token is invalid or user is already activate."));
+
   }
```

Comment:
Con lo anterior esto se solucionaría. Acá estás (1) buscando la ID del usuario que ya buscaste previamente y (2) buscando ese mismo usuario por la id. Son dos consultas iguales a la base de datos. Cambiémoslo por esto
```suggestion
  private User findUserToActivateOrThrowException(UUID activationTokenUUID) {
    return tokenService.findUserByActivationToken(activationTokenUUID); 
    // A la excepción la manejás en el token service
  }
```